tests/main/services-restart/task.yaml

```diff
@@ -0,0 +1,69 @@
+summary: Check that snap restarts affects only running services
+
+execute: |
+    echo "Installing the service snap"
+
+    # we are using systemd-notify indicate the service is active, this is
+    # currently not allowed by daemon-notify interface, so we may as well just
+    # install in devmode
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-service-restart --devmode
+
+    echo "We can see all services running"
+    for id in 1 2 3 4; do
+        systemctl status snap.test-snapd-service-restart.svc${id}.service | MATCH "running"
+    done
+
+    echo "Stopping services 3 and 4"
+    snap stop test-snapd-service-restart.svc3
+    snap stop test-snapd-service-restart.svc4
+
+    echo "Disabling services 1 and 3"
+    systemctl disable snap.test-snapd-service-restart.svc1.service
+    systemctl disable snap.test-snapd-service-restart.svc3.service
+
+    # At this point, we have:
+    # 1 disabled service that is running (svc1)
+    # 1 enabled service that is running (svc2)
+    # 1 disabled service that is stopped (svc3)
+    # 1 enabled service that is stopped (svc4)
```

Comment:
```suggestion
    # svc2 as disabled service that is running
    # svc2 as enabled service that is running
    # svc3 as disabled service that is stopped
    # svc4 as enabled service that is stopped
```

Reply:
thanks for this comment btw it is very helpful in figuring out the spread test's state of things

Reply:
Done